### ISSUE TITLE
Client disconnect when opening two instances within one Docker container

### ISSUE THREAD

**Issue by BladeMF** [microsoft/vscode-remote-release#8575]
Does this issue occur when all extensions are disabled?: Yes

**VSCode version**
Version: 1.74.0 (user setup)
Commit: 5235c6bb189b60b01b1f49062f4ffa42384f8c91
Date: 2022-12-05T16:38:16.075Z
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: No

**OS version**
Version 22H2 (OS Build 22621.900)

**Steps to Reproduce:**

1. Open a project in [Dev Container] within a Docker container
2. Open a second project in [Dev Container] within _the same_ Docker container
3. The first project's client disconnects after a short period

The `Window` log of the disconnected instance says:
```
2022-12-13 10:07:15.232 [info] [remote-connection][Management   ][9b5ab…][reconnect] received socket close event (wasClean: false, code: 1006, reason: ).
2022-12-13 10:07:15.234 [error] {"isTrusted":true}
2022-12-13 10:07:15.235 [info] [remote-connection][Management   ][9b5ab…][reconnect] starting reconnecting loop. You can get more information with the trace log level.
2022-12-13 10:07:15.235 [info] [remote-connection][Management   ][9b5ab…][reconnect] resolving connection...
2022-12-13 10:07:15.237 [info] Invoking resolveAuthority(dev-container)
2022-12-13 10:07:15.239 [info] [remote-connection][ExtensionHost][67da3…][reconnect] received socket close event (wasClean: false, code: 1006, reason: ).
2022-12-13 10:07:15.239 [error] {"isTrusted":true}
2022-12-13 10:07:15.239 [info] [remote-connection][ExtensionHost][67da3…][reconnect] starting reconnecting loop. You can get more information with the trace log level.
2022-12-13 10:07:15.240 [info] [remote-connection][ExtensionHost][67da3…][reconnect] resolving connection...
2022-12-13 10:07:15.404 [info] resolveAuthority(dev-container) returned '127.0.0.1:42559' after 169 ms
2022-12-13 10:07:15.404 [info] [remote-connection][Management   ][9b5ab…][reconnect] connecting to 127.0.0.1:42559...
2022-12-13 10:07:15.405 [info] [remote-connection][ExtensionHost][67da3…][reconnect] connecting to 127.0.0.1:42559...
2022-12-13 10:07:15.831 [error] [remote-connection][Management   ][9b5ab…][reconnect][127.0.0.1:42559] socketFactory.connect() failed or timed out. Error:
2022-12-13 10:07:15.831 [error] CodeExpectedError: WebSocket close with status code 1006
    at WebSocket.<anonymous> (vscode-file://vscode-app/c:/Users/vladi/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:641:37616)
2022-12-13 10:07:15.831 [info] [remote-connection][Management   ][9b5ab…][reconnect] A temporarily not available error occurred while trying to reconnect, will try again...
2022-12-13 10:07:15.832 [info] [remote-connection][Management   ][9b5ab…][reconnect] waiting for 5 seconds before reconnecting...
```

The second instance's `exthost.log` file says:
```
2022-12-13 08:14:25.106 [info] Extension host with pid 22992 started
2022-12-13 08:14:25.110 [error] Error: EEXIST: file already exists, open '/root/.vscode-server/data/User/workspaceStorage/c766257cd5d10e8c872a327d311edb7d/vscode.lock'
2022-12-13 08:14:25.116 [info] Lock '/root/.vscode-server/data/User/workspaceStorage/c766257cd5d10e8c872a327d311edb7d/vscode.lock': Could not acquire lock, checking if the file is stale.
2022-12-13 08:14:25.126 [info] Lock '/root/.vscode-server/data/User/workspaceStorage/c766257cd5d10e8c872a327d311edb7d/vscode.lock': The pid 19754 appears to be gone.
2022-12-13 08:14:25.126 [info] Lock '/root/.vscode-server/data/User/workspaceStorage/c766257cd5d10e8c872a327d311edb7d/vscode.lock': Deleting a stale lock.
2022-12-13 08:14:25.133 [info] Lock '/root/.vscode-server/data/User/workspaceStorage/c766257cd5d10e8c872a327d311edb7d/vscode.lock': Lock acquired.
2022-12-13 08:14:25.231 [info] ExtensionService#_doActivateExtension vscode.configuration-editing, startup: false, activationEvent: 'onLanguage:json'
2022-12-13 08:14:25.233 [info] ExtensionService#_doActivateExtension vscode.extension-editing, startup: false, activationEvent: 'onLanguage:json'
2022-12-13 08:14:25.233 [info] ExtensionService#_doActivateExtension vscode.json-language-features, startup: false, activationEvent: 'onLanguage:json'
2022-12-13 08:14:25.233 [info] ExtensionService#_doActivateExtension vscode.npm, startup: false, activationEvent: 'onLanguage:json'
2022-12-13 08:14:25.234 [info] ExtensionService#_doActivateExtension vscode.git-base, startup: true, activationEvent: '*', root cause: vscode.git
2022-12-13 08:14:25.416 [info] ExtensionService#_doActivateExtension vscode.git, startup: true, activationEvent: '*'
2022-12-13 08:14:25.416 [info] ExtensionService#_doActivateExtension vscode.github, startup: true, activationEvent: '*'
2022-12-13 08:14:25.619 [info] ExtensionService#_doActivateExtension vscode.debug-auto-launch, startup: false, activationEvent: 'onStartupFinished'
2022-12-13 08:14:25.619 [info] ExtensionService#_doActivateExtension vscode.emmet, startup: false, activationEvent: 'onStartupFinished'
2022-12-13 08:14:25.619 [info] ExtensionService#_doActivateExtension vscode.merge-conflict, startup: false, activationEvent: 'onStartupFinished'
2022-12-13 08:14:25.620 [info] Eager extensions activated
```

**Comment by chrmarti** [comment]
@BladeMF Thanks for sending the logs. They indicate that the second window terminates the agent running in the container before starting a new one. Could you enable trace logging for Dev Containers in the user settings, retry and send me the logs from both windows again?
<img width="501" alt="image" src="https://github.com/microsoft/vscode-remote-release/assets/9205389/052c51b4-f548-40b7-9ada-5a0f34a18255">

**Comment by BladeMF** [comment]
Sure, I'll try and do that today.

Vladi.
________________________________

**Comment by BladeMF** [comment]
I've actually forgotten about this. The issue still persists. I'll try and do the logs next week.

**Comment by BladeMF** [comment]
@chrmarti , I've collected the logs. So, the situation seems changed:
1. Open one folder with Reopen in Container. All good.
2. Open a second folder in the same container with Reopen in Container. All good. No matter how many times I refreshed that window the first one did not disconnect. That's an improvement
3. In the second folder I used "Attach to running container" and selecting the same container in order to open another folder in the same contrainer which immediately disconnected the first instance. Reloading the first instance disconnected the second again. Reloading the second reset it back to the original folder which I opened and they were both stable.

I will email you the logs from the two instances.
